modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -26,7 +26,7 @@ package object engine {
     * without interruption. A *sequential* `Execution` can be represented with
     * an `Execution` with a single `Action`.
     */
-  type Execution[A] = NonEmptyList[A]
+  type Execution[A] = List[A]
```

Comment:
The reason for replacing `nonEmptyList`s by `List`s all over the code is that we should be able to carry out an actual execution independently at each level of the state hierarchy (Queue -> Sequence -> Step), so the empty case represents the finalization.

However in the case of `Execution` it should still make sense to keep it as `NonEmptyList` since it only can move to the next execution when all actions are completed, not when it's empty. I'll try to change it back.